lib/OpenTracing/Implementation/DataDog/Tracer.pm

```diff
@@ -269,10 +274,28 @@ sub inject_context_into_hash_reference   {
 
 
 
-sub inject_context_into_http_headers     { return $_[1] } # $carrier
```

Comment:
I wish I had done this more correctly and had written this as:

```perl
return $_[1]->clone
```

At least we would have the correct starting point

See: [OT::Interface::Tracer inject_context](https://metacpan.org/pod/OpenTracing::Interface::Tracer#inject_context), and the topic of immutable objects (and functional programming)

Is it worth going over other `inject_context` methods across our own implementations and see that we indeed used `clone` or other means to make a (shallow) clone ?

Reply:
Good point, I've added a clone there. The only other inject is the hash one and that is being cloned courtesy of Hash::Merge, so I think we're good